packages/devtools_app/lib/src/screens/memory/panes/diff/controller/diff_pane_controller.dart

```diff
@@ -146,6 +143,28 @@ class DiffPaneController extends DisposableController {
       ),
     );
   }
+
+  late final _classFilterController = ClassFilterController(
+    filter: core.classFilter,
+    onChanged: applyFilter,
+    rootPackage: serviceManager.rootInfoNow().package,
+  );
+
+  late final ClassesTableSingleController classesTableSingleController =
+      ClassesTableSingleController(
+    selection: derived.selectedSingleClassStats,
```

Comment:
instead of passing in the selection notifier, define this notifier directly in the `ClassesTableSingleController` class. That way we aren't duplicating the notifiers and it can only be modified through the `ClassesTableSingleController` class.

 Same comment for `ClassesTableDiffController`'s selection notifier as well